packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -1525,8 +1525,8 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
 
     // Validate against instance type restrictions, per
     // https://docs.aws.amazon.com/opensearch-service/latest/developerguide/supported-instance-types.html
-    if (isSomeInstanceType('i3', 'r6gd') && ebsEnabled) {
-      throw new Error('I3 and R6GD instance types do not support EBS storage volumes.');
+    if (isSomeInstanceType('r3', 'i3', 'r6gd', 'im4gn') && ebsEnabled) {
+      throw new Error('R3, I3, R6GD and IM4GN instance types do not support EBS storage volumes.');
```

Comment:
this `if` was missing `r3`

Reply:
The docs linked above say that `r3` is incompatible with encryption at rest but is silent about EBS specifically. Are the service docs missing that fact, is there another source for that limitation? https://docs.aws.amazon.com/opensearch-service/latest/developerguide/supported-instance-types.html

Reply:
I _think_ that doc is missing it? On the [pricing page](https://aws.amazon.com/opensearch-service/pricing/), it doesn't have EBS and is similar to `i3`, `r6gd`, and `im4gn`.

Reply:
@kylelaker Let me fix this. I tested in the Console and you are allowed to have EBS for R3.

Reply:
@kylelaker Thanks for catching this! I confirmed `r3` supports both instance and EBS.